common/data_refinery_common/models/models.py

```diff
@@ -305,6 +305,16 @@ def to_metadata_dict(self):
 
         return metadata
 
+    def get_sample_metadata_fields(self):
+        """ Get all metadata fields that are non-empty for at least one sample in the experiment.
+        See https://github.com/AlexsLemonade/refinebio-frontend/issues/211 for why this is needed.
+        """
+        fields = []
+        for sample in self.samples.all():
+            metadata = sample.to_metadata_dict()
+            fields += [k for k, v in metadata.items() if v and k not in fields]
```

Comment:
> if v and k not in fields

Are you sure we should be checking for both v and k? Seems like that would cause the key to be added once for every different value it has.

Reply:
`not in` has higher precedence than `and`, so with parenthesis the comparison would be `if (v) and (k not in fields)`, which is correct. It's checking if `v` is truthy and `k` is not already in `fields`.

Reply:
I'm also not crazy about this bit, surely this should be doable entirely at the query level so we don't need to loop at all?

Reply:
For my part:

Duh, of course that's how that logic is working... Sorry 'bout that.